src/commands/dev/socket.rs

```diff
@@ -19,66 +30,309 @@ use tokio_tungstenite::{connect_async, tungstenite, MaybeTlsStream, WebSocketStr
 use anyhow::{anyhow, Result};
 use url::Url;
 
+use super::ServerConfig;
+
 const KEEP_ALIVE_INTERVAL: u64 = 10;
+const DEVTOOLS_PORT: u16 = 9230;
 
 /// connect to a Workers runtime WebSocket emitting the Chrome Devtools Protocol
 /// parse all console messages, and print them to stdout
-pub async fn listen(socket_url: Url) -> Result<()> {
+///
+/// `inspect` should be the name of the worker if `--inspect` is passed, or `None` otherwise.
+pub async fn listen(
+    socket_url: Url,
+    server_config: ServerConfig,
+    inspect: Option<String>,
+) -> Result<()> {
     // we loop here so we can issue a reconnect when something
     // goes wrong with the websocket connection
     loop {
-        let ws_stream = connect_retry(&socket_url).await;
+        // TODO: replace this with a prewarm call
+        if inspect.is_some() {
+            // Startup the edge isolate.
+            // TODO: does this need to be https in some cases?
+            // let scheme = if server_config.host.is_https() { "https" } else { "http" };
 
-        let (mut write, read) = ws_stream.split();
+            // Sleep for a bit first to give the webserver time to start up.
+            tokio::time::sleep(Duration::from_millis(200)).await;
+            connect_retry(|| reqwest::get(format!("http://{}", server_config.listening_address)))
+                .await;
+        }
+        let mut ws_stream = connect_retry(|| connect_async(&socket_url)).await.0;
 
         // console.log messages are in the Runtime domain
         // we must signal that we want to receive messages from the Runtime domain
         // before they will be sent
         let enable_runtime = protocol::runtime::SendMethod::Enable(1.into());
         let enable_runtime = serde_json::to_string(&enable_runtime)?;
         let enable_runtime = tungstenite::protocol::Message::Text(enable_runtime);
-        write.send(enable_runtime).await?;
+        ws_stream.send(enable_runtime).await?;
 
-        // if left unattended, the preview service will kill the socket
-        // that emits console messages
-        // send a keep alive message every so often in the background
-        let (keep_alive_tx, keep_alive_rx) = mpsc::unbounded_channel();
+        // parse all incoming messages and print them to stdout
+        if let Some(worker_name) = &inspect {
+            StdErr::help("Open chrome://inspect, click 'Configure', and add localhost:9230");
+            // StdErr::help(&format!("Open devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=localhost:{port}{path} in Chrome", port=DEVTOOLS_PORT, path=socket_url.path()));
 
-        // every 10 seconds, send a keep alive message on the channel
-        let heartbeat = keep_alive(keep_alive_tx);
+            // Construct our SocketAddr to listen on...
+            let addr = SocketAddr::from(([127, 0, 0, 1], DEVTOOLS_PORT));
```

Comment:
Instead of hard-coding a port, can we just pass port `0` and allow the OS to allocate one for us? `wrangler tail` has had issues with that.

Reply:
Hmm, that would be a little more work for the end user, because they'd have to change the port Chrome looks at each time they restart Wrangler.

What trouble has `tail` had?

Reply:
Ah okay, if Chrome uses a stable port this is fine. There was an issue with port conflict with the old `tail` that uses `cloudflared`. Issue was here: https://github.com/cloudflare/wrangler/issues/1965

Reply:
Yup, node and deno both use 9229, so this uses 9230 to avoid conflicting. I think Chrome already has 9222 enabled by default, so if we used that it would be one less step, but I'm not sure what already uses that port.